src/Compilers/Core/Portable/Symbols/Attributes/CommonAttributeData.cs

```diff
@@ -292,7 +293,47 @@ private ObsoleteAttributeData DecodeObsoleteAttribute()
                 }
             }
 
-            return new ObsoleteAttributeData(ObsoleteAttributeKind.Obsolete, message, isError);
+            Debug.Assert(AttributeClass is object);
+
+            string? diagnosticId = null;
+            string? urlFormat = null;
+            foreach (var pair in this.CommonNamedArguments)
+            {
+                switch (pair.Key)
+                {
+                    case "DiagnosticId" when isProperty("DiagnosticId"):
+                        diagnosticId = pair.Value.ValueInternal as string;
+                        break;
+                    case "UrlFormat" when isProperty("UrlFormat"):
+                        urlFormat = pair.Value.ValueInternal as string;
```

Comment:
>urlFormat = pair.Value.ValueInternal as string; [](start = 24, length = 47)

Consider if we should skip it if we already have the value.

Reply:
I went with skipping it when we already have the value and added an IL test with multiple arguments for each property.

Reply:
> I went with skipping it when we already have the value and added an IL test with multiple arguments for each property.

I see this done for PE, but it doesn't look like we are doing this for source. Should also have a test for source, even though there will be a CS0643 error.

---
In reply to: [398252689](https://github.com/dotnet/roslyn/pull/42518#discussion_r398252689) [](ancestors = 398252689)

Reply:
Interesting, I expected no obsolete diagnostic to be given in this scenario due to errors on the attribute, but it looks like we do give a diagnostic. Will add the test.